netutils/nist.py

```diff
@@ -0,0 +1,130 @@
+"""Functions building NIST URLs from the os platform values."""
+import re
+import typing as t
+
+
+def get_nist_urls_juniper_junos(os_platform_data: t.Dict[str, t.Any], api_key: str) -> t.List[str]:
+    """Create a list of possible NIST Url strings for JuniperPlatform.
+
+    Args:
+        api_key: NIST-API-KEY - Request here https://nvd.nist.gov/developers/request-an-api-key
+
+    Returns:
+        List of NIST CPE URLs that may contain platform data.
+    """
+    nist_urls = []
+    base_url = f"""https://services.nvd.nist.gov/rest/json/cpes/1.0?apiKey={api_key}&addOns=cves&cpeMatchString=cpe:2.3:o:juniper:junos"""
+
+    url_args = {"base_url": base_url, **os_platform_data}
```

Comment:
Can you explain to me why `url_args` as a dictionary is beneficial? in my mind we can limit the line length a fair bit with this strategy. 

```python
delim_six = ":*" * 6
delim_seven = ":*" * 7
base_ext = # use if conditional logic 
if os_platform_data["isspecial"]:
    _type = os_platform_data["type]

base_type_build = {base_ext}{_type}{build}
```

Reply:
That technique I believe was brought in by @progala for "object factory".

It worked to satisfy Jeff's dataclasses request and I didn't question it, this is not yet an area I'm knowledgeable about.

Przemek, would you be able to elaborate for Ken, or would his suggestion be in our best interest?

Reply:
@bminnix `os_platform_data` is the one built by object factory. If we can limit the line length by getting rid of `url_args` then it's worth trying.

Reply:
@itdependsnetworks 
I have made changes to shorten the line lengths.